spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -352,4 +355,16 @@ public Catalog icebergCatalog() {
         "Cannot return underlying Iceberg Catalog, wrapped catalog does not contain an Iceberg Catalog");
     return ((HasIcebergCatalog) icebergCatalog).icebergCatalog();
   }
+
+  @Override
+  public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionException {
+    try {
+      return super.loadFunction(ident);
+    } catch (NoSuchFunctionException e) {
+      if (getSessionCatalog() instanceof FunctionCatalog) {
```

Comment:
Question: Is it possible for this not to be a FunctionCatalog?

Reply:
It's a guard condition before trying to cast the `getSessionCatalog` into `FunctionCatalog`. And `loadFunction` is method of Spark's `FunctionCatalog`.

Reply:
Since SPARK-37731(https://github.com/apache/spark/pull/35004, fixed in 3.3.0), `V2SessionCatalog` extends `FunctionCatalog`

```patch
class V2SessionCatalog(catalog: SessionCatalog)
-  extends TableCatalog with SupportsNamespaces with SQLConfHelper {
+  extends TableCatalog with FunctionCatalog with SupportsNamespaces with SQLConfHelper {
```

https://github.com/apache/spark/pull/35004/files#diff-2d6f351fff8241ff1187b98a62e6c57ef3b55349658a9eb98056a14c51a9dc7cL40-R43

Reply:
Yeah my main thought here is can this guard ever be false, I don't think it can but I don't have an issue with keeping the check.

Reply:
I keep the cheek for Spark 3.2, but skipped in Spark 3.3 as V2SessionCatalog extends `FunctionCatalog `.
Thanks for the hints from @pan3793 .